apps/s_server.c

```diff
@@ -595,39 +640,53 @@ static int cert_status_cb(SSL *s, void *arg)
 {
     tlsextstatusctx *srctx = arg;
     OCSP_RESPONSE *resp = NULL;
-    unsigned char *rspder = NULL;
-    int rspderlen;
+    STACK_OF(OCSP_RESPONSE) *sk_resp = NULL;
+    char* respfile = NULL;
     int ret = SSL_TLSEXT_ERR_ALERT_FATAL;
+    int i;
+    BIO *derbio;
 
     if (srctx->verbose)
         BIO_puts(bio_err, "cert_status: callback called\n");
 
-    if (srctx->respin != NULL) {
-        BIO *derbio = bio_open_default(srctx->respin, 'r', FORMAT_ASN1);
-        if (derbio == NULL) {
-            BIO_puts(bio_err, "cert_status: Cannot open OCSP response file\n");
-            goto err;
-        }
-        resp = d2i_OCSP_RESPONSE_bio(derbio, NULL);
-        BIO_free(derbio);
-        if (resp == NULL) {
-            BIO_puts(bio_err, "cert_status: Error reading OCSP response\n");
-            goto err;
+    if (srctx->skrespin != NULL) {
+        for (i = 0; i < sk_OPENSSL_STRING_num(srctx->skrespin); i++) {
+            respfile = sk_OPENSSL_STRING_value(srctx->skrespin, i);
```

Comment:
Since `respfile` may now contain a whole list of OCSP responses, you must extend the respective help output for `-status_file"` and document this also in `openssl_s_server.pod.in`.

Reply:
Would be that ok

>     {"status_file", OPT_STATUS_FILE, '<',
>      "File containing DER encoded OCSP Response (can be specified multiple times)"},

and for openssl_s_server.pod.in:

> Overrides any OCSP responder URLs from the certificate and always provides the
> OCSP Response stored in the file. The file must be in DER format.
> This option may be used multiple times to specify OCSP Responses for all
> certificates in the server certificate chain.